upgrade/upgrade-scripts/execute

```diff
@@ -350,12 +350,12 @@ zcat "/usr/share/doc/delphix-entire-$platform/packages.list.gz" |
 # First, since "apt-get autoremove" does not work properly for the
 # kernel packages, we have to handle the removal of those here.
 #
-KVERS=$(apt-mark showauto 'delphix-kernel-.*' | sed 's/^delphix-kernel-//') ||
-	die "failed to determine list of kernel versions to remove"
-for KVER in $KVERS; do
-	apt_get purge -y "delphix-kernel-$KVER" "linux-image-$KVER" ||
-		die "failed to remove kernel packages for version '$KVER'"
-done
+apt-mark showauto 'delphix-kernel-.*' | while read -r pkg; do
+	# shellcheck disable=SC2046
+	apt_get purge -y "$pkg" \
+		$(dpkg-query -Wf '${Depends}' "$pkg" | sed 's/, / /g') ||
```

Comment:
Further, as a result of that call, I see it removing more than just `delphix-kernel-5.0.0-29-generic` packages:
```
Jun  9 17:47:35 ps-64201-8 upgrade-scripts:execute[19743]: The following packages will be REMOVED:
Jun  9 17:47:35 ps-64201-8 upgrade-scripts:execute[19743]:   connstat-module-5.0.0-29-generic* delphix-kernel-5.0.0-29-generic*
Jun  9 17:47:35 ps-64201-8 upgrade-scripts:execute[19743]:   delphix-kernel-5.0.0-37-generic* delphix-kernel-5.3.0-53-generic*
Jun  9 17:47:35 ps-64201-8 upgrade-scripts:execute[19743]:   linux-generic-hwe-18.04* linux-image-5.0.0-29-generic*                             
Jun  9 17:47:35 ps-64201-8 upgrade-scripts:execute[19743]:   linux-modules-extra-5.0.0-29-generic* linux-tools-generic-hwe-18.04*                                                                                                                                                              
Jun  9 17:47:35 ps-64201-8 upgrade-scripts:execute[19743]:   zfs-headers-5.0.0-29-generic* zfs-modules-5.0.0-29-generic*    
```
I see it removing `delphix-kernel-5.0.0-37-generic` here as well.. I'm not sure that's the behavior we want.

i.e. in this case, both `delphix-kernel-5.0.0-29-generic` and `delphix-kernel-5.0.0-37-generic` depend on `linux-generic-hwe-18.04 linux-tools-generic-hwe-18.04`.. and since we're removing `linux-generic-hwe-18.04 linux-tools-generic-hwe-18.04`, both `delphix-kernel` packages are removed...

Reply:
Immediately after that, the script will explicitly try to remove `delphix-kernel-5.0.0-37-generic`, which fails because it was already implicitly removed:
```
Jun  9 17:51:16 ps-64201-8 upgrade-scripts:execute[19743]: /var/dlpx-update/latest/common.sh:apt_get:341 apt-get -o Dpkg::Options::=--force-confdef -o Dpkg::Options::=--force-confold purge -y delphix-kernel-5.0.0-37-generic
Jun  9 17:51:17 ps-64201-8 upgrade-scripts:execute[19743]: Reading package lists...         
Jun  9 17:51:17 ps-64201-8 upgrade-scripts:execute[19743]: Building dependency tree...                        
Jun  9 17:51:17 ps-64201-8 upgrade-scripts:execute[19743]: Reading state information...                    
Jun  9 17:51:17 ps-64201-8 upgrade-scripts:execute[19743]: E: Unable to locate package delphix-kernel-5.0.0-37-generic
Jun  9 17:51:17 ps-64201-8 upgrade-scripts:execute[19743]: E: Couldn't find any package by glob 'delphix-kernel-5.0.0-37-generic'  
Jun  9 17:51:17 ps-64201-8 upgrade-scripts:execute[19743]: E: Couldn't find any package by regex 'delphix-kernel-5.0.0-37-generic'  
```
Since older versions of `delphix-kernel` depend on non-kernel versioned packages, I think we should be explicit here in which packages we remove.. which is what I did in the first version of this PR.

Either that, or some how filter out the non-kernel versioned dependencies from the dynamically generated list.. such that we only attempt to remove the packages that have the kernel version in the name.

Reply:
I still need to test this, but I think something like this might work better for generating the dependent packages to remove:
```
$ KVERS=5.0.0-29-generic
$ apt-cache depends delphix-kernel-$KVERS | awk '$1 == "Depends:" {print $2}' | grep -E ".*-$KVERS"
zfs-modules-5.0.0-29-generic
zfs-headers-5.0.0-29-generic
linux-image-5.0.0-29-generic
connstat-module-5.0.0-29-generic
```
thoughts?

Reply:
Yeah, we would have to rely on the running version rather than installed version.
When deleting delphix-kernel versions generated prior to 6.0.6.0 we may want to be explicit in what packages we remove. Feel free to reach out if you want to discuss this offline.